logger/honeycomb.go

```diff
@@ -59,7 +59,7 @@ func (h *HoneycombLogger) Start() error {
 		}
 	}
 
-	if loggerConfig.SamplerEnabled {
+	if *loggerConfig.SamplerEnabled {
```

Comment:
We should consider adding a getter function for HoneycombLogger.SamplerEnabled so that we do not have to dereference in every time Sampler Enabled is used.

Reply:
During a chat this morning, @kentquirk, @fchikwekwe, and I agreed we should make the SamplerEnabled bool-pointer private and add a getter function to smooth the bool interface. That'll be work on this PR with other ideas to reduce tech debt in a follow-up issue.